lib/superagent-mock.js

```diff
@@ -47,6 +47,45 @@ function mock (superagent, config, logger) {
   }();
 
   /**
+   * Chains progrss events and eventually returns the response
+   * @param ProgressEvent (object/function) the ProgressEvent implementation
+   * @param request (object) the current request
+   *
+   * @param progress (object) the progress configuration
+   * @param progress.parts (number)
+   * @param progress.delay (number)
+   * @param progress.total (number)
+   * @param progress.lengthComputable (boolean)
+   * @param progress.direction (string)
+   *
+   * @param remainingParts (number) the remaining calls to make
+   * @param callback (function) the end response callback
+   */
+  var handleProgress = function (ProgressEvent, request, progress, remainingParts, callback) {
+    setTimeout(function () {
+      var total = progress.total || 100;
+      var loaded = total * (progress.parts - remainingParts) / progress.parts;
+      var lengthComputable = typeof progress.lengthComputable !== 'undefined' ? !!progress.lengthComputable : true;
+      var e = ProgressEvent.initProgressEvent
```

Comment:
You asked about the benefit of using the ProgressEvent api. I would say that I wanted to mock exactly the event that would occur if a real progress event happened so that the developer wouldn't have to worry about it in their code. It was my first thought, but we could go any other way and it will work the same.